test/test_rb.py

```diff
@@ -1553,6 +1559,53 @@ def test_error_noninit(self):
 
 
 class TestSamplers:
+    @pytest.mark.parametrize("backend", ["torch", "torchsnapshot"])
+    def test_sampler_without_rep_state_dict(self, backend):
+        if backend == "torchsnapshot" and not _has_snapshot:
+            pytest.skip()
```

Comment:
Otherwise you could make the backend list dependent of whether _has_torchsnapshot is true of false

Reply:
if you prefer that...

In that case it doesn't show that there were tests skipped and the number of tests will vary depending on the pipeline

Reply:
Yes but that's better, for instance fbcode does not like skipping tests 
Better not have a test listed than skipped systematically